### PR TITLE
FLA-1100 Add Keycloak

### PR DESCRIPTION
**Pull request by BrianMaki**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+MVN_PROFILE=
+SERVER_PORT=8090
+
+# KEYCLOAK CONFIG
+KEYCLOAK_SSL_REQUIRED=none
+KEYCLOAK_RESOURCE=ai-reviewer-api
+KEYCLOAK_URL=http://localhost:8081/auth
+KEYCLOAK_REALM=Efiling-Reviewer-Hub
```
Comment (KDNeufeld) [inline .env.template:8]:
```suggestion
KEYCLOAK_REALM=ai-reviewer
```

File: docker-compose.yml [hunk 2]
```diff
@@ -41,6 +41,11 @@ services:
       - CLAMAV_TIMEOUT=50000
       - SW_AGENT_NAME=ai-reviewer-api
       - SW_AGENT_COLLECTOR_BACKEND_SERVICES=skywalking-oap-server:11800
+      - KEYCLOAK_AUTH_SERVER_URL=${KEYCLOAK_AUTH_SERVER_URL:-http://keycloak:8080/auth}
+      - KEYCLOAK_REALM=${KEYCLOAK_REALM:-Efiling-Reviewer-Hub}
```
Comment (KDNeufeld) [inline docker-compose.yml:45]:
```suggestion
      - KEYCLOAK_REALM=${KEYCLOAK_REALM:-ai-reviewer}
```

File: src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/config/KeycloakSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package ca.bc.gov.open.jag.efilingreviewerapi.config;
```
Comment (KDNeufeld) [inline src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/config/KeycloakSecurityConfig.java:1]:
```suggestion
package ca.bc.gov.open.jag.aireviewerapi.config;
```

**Comment by BrianMaki** [comment]
Created a new branch to simplify PR.

**Pull request closed without merging**